City-of-Helsinki/drupal-hdbt
# UHF-2976: Clean regions

# Clean regions [UHF-2976](https://helsinkisolutionoffice.atlassian.net/browse/UHF-2976)

## What was done
* Breadcrumbs were moved inside main region as [a11y experts noted](https://helsinkisolutionoffice.atlassian.net/wiki/spaces/HEL/pages/7225048911/19.5+saavutettavuusauditoinnin+ty+lista) that it's better for them to be there.
* The old sidebar_first was split to sidebar_first and sidebar_second
  * Source order (and tab order) is now sidebar_first main_content sidebar_second
  * Sidebar_first contains only subnavigation
  * Sidebar_second contains only additional page related information
  * This was the main reason for this whole operation (together with code refactoring and cleanup)
* TPR pages sidebars were moved from content area to actual sidebar areas for unity
* Moved page.messages area below skip-to-content anchor for accessibility
* Promoted sidebar title to h2 from div
* Unified basic page, news page, tpr-service and tpr-unit content to use same code with same features
  * page-title works the same (always before the article element, when there is no hero)
  * lead-in (previously known as lead-in and short-desc)
  * page-divider uses same code, visible always when there is no ToC
  * ToC uses same code
  * main-image (merged news and unit image styles to use same base code)
  * long-desc (where used, behaves well with the above)
  * enriched-content (margin things for the new layout)
* Page layout uses now css grid to handle subnav, sidebar and upper content
* Fixed a bunch of scss linting errors (still plenty to go)
* Removed gallery code as it was causing problems with new layout and was due for removal in [UHF-4069](https://helsinkisolutionoffice.atlassian.net/browse/UHF-4069) anyway
* Changed page layout to have fixed width sidebar of 340px, with gap of 64px and in its widest form, main content at 860px. (This was done in co-operation with Sauli)
* Mobile menu checkbox was causing accessibility warnings as it was changed to be of role button, so a proper label was added for it.
* Fixed a content liftup image bug, where a broken image would also break the page layout.


## How to install
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin dev`
    * `make fresh`
* If you're using **Sote** or **Kasko**, follow the instructions below
    * Checkout branch `UHF-2976_clean_regions`
    * `make drush-cr`
* Update the necessary modules and themes
    * `composer require drupal/hdbt:dev-UHF-2976_clean_regions && composer require drupal/helfi_tpr:dev-UHF-2976_clean_regions && composer require drupal/helfi_platform_config:dev-UHF-2976_clean_regions`
* Run `make drush-updb drush-cr`

## How to test
* Go to any landing page and check that the layout doesn't break. Compare it to landing pages in production
* Go to a standard page of your choice. Check that following works and the layout doesn't break
   - [ ] The page is in menu and it doesn't have sidebar content
       * The layout should have menu sidebar and it [should look like this](https://user-images.githubusercontent.com/1712902/168118439-519a29a8-abb0-4b64-be63-a67f41146bdf.png)
   - [ ] The page is in menu and it has sidebar content
       * The layout should have menu sidebar, there should be sidebar content underneath menu sidebar and [it should look like this](https://user-images.githubusercontent.com/1712902/168117962-198ddc52-4886-4d38-ab6e-33a3caf0bba3.png)
   - [ ] The page is not in menu OR the "Hide navigation from this page" is checked, AND it has sidebar content
       * The layout should have sidebar content (on the right side) and it [should look like this](https://user-images.githubusercontent.com/1712902/168117337-8034c4f3-9c34-4fef-95f1-a947a2e624a9.png)
   - [ ] The page is not in menu OR the "Hide navigation from this page" is checked AND it doesn't have sidebar content
       * The layout shouldn't have sidebars and it [should look like this](https://user-images.githubusercontent.com/1712902/168118654-f0715316-b33a-4277-9842-8c5373cf6ac0.png)
* Test the TPR Unit and TPR Service pages with similar steps as above. The layouts should be nearly identical to standard page layouts. 
   - There was a big overhaul with the TPR unit and service templates. For example sidebar content, sidebar menu, contact information and page title has been moved to blocks. **So be extra careful when testing units and services and try your best to break them.**  

## Other PRs
* https://github.com/City-of-Helsinki/drupal-helfi-sote/pull/366
* https://github.com/City-of-Helsinki/drupal-helfi-platform/pull/64
* https://github.com/City-of-Helsinki/drupal-helfi-kasvatus-koulutus/pull/148
* https://github.com/City-of-Helsinki/drupal-helfi-platform-config/pull/315
* https://github.com/City-of-Helsinki/drupal-module-helfi-tpr/pull/118
